File: leetcode/strings/LengthOfLastWord.java
/*
58. Length of Last Word
Solved
Easy
Topics
Companies
Given a string s consisting of words and spaces, return the length of the last word in the string.

A word is a maximal substring consisting of non-space characters only.
 */

 class LengthOfLastWord {
    private static int lengthOfLastWord(String s) {
        int idx = s.length() - 1;
        int len = 0;

        // Skip all whitespaces at the end of 
        while(idx>=0 && s.charAt(idx) == ' ') idx--; 
        // Count the number of consecutive non-whitespace characters
        while(idx>=0 && s.charAt(idx) != ' ') {idx--; len++;}
        return len;
    }

    public static void main(String[] args) {
    	String s = "   fly me   to   the moon  ";
    	System.out.println(lengthOfLastWord(s));
    }
}

File: leetcode/arrays/straightforward/length_of_last_word.py
"""
58. Length of Last Word
Solved
Easy
Topics
Companies
Given a string s consisting of words and spaces, return the length of the last word in the string.

A word is a maximal 
substring
 consisting of non-space characters only.
"""


class LengthOfLastWord:
    def lengthOfLastWord(self, s: str) -> int:
        # return len(s.split()[-1])
        lenLast = 0

        for i in s[::-1]:
            if i == " ":
                if lenLast > 0:
                    return lenLast
            else:
                lenLast += 1

        return lenLast
